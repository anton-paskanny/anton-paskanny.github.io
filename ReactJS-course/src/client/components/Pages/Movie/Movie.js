import React from 'react';

import HeaderDetail from '../../Header/HeaderDetail';
import MovieResults from '../../../containers/MovieResults';

const Movie = ({ match }) => (
    <>
        <HeaderDetail movieId={match.params.id} />
        <MovieResults movieDetailPage={true} />
    </>
);

export default Movie;
