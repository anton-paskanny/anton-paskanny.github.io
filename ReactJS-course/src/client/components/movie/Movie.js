import React from 'react';
import { Link } from 'react-router-dom';

import './styles.css';

const Movie = (props) => {
  const {
    id, title, genres, poster_path, release_date,
  } = props.data;

  function renderGenres(genres) {
    return genres && genres.map((title, index) => (
      index > 0
        ? <React.Fragment key={title}>
                <li className='movie__genre-list-item--separator'>/</li>
                <li className='movie__genre-list-item'>{title}</li>
            </React.Fragment>
        : <li key={title} className='movie__genre-list-item'>{title}</li>
    ));
  }

  return (
        <Link to={`/movie/${id}`} className="movie">
            <figure className="movie__img-placeholder">
                <img className="movie__img"src={poster_path} alt={title}/>
            </figure>
            <div className="movie__info-wrapper">
                <div className="movie__desc-wrapper">
                    <h3 className="movie__title">{title}</h3>
                    <ul className="movie__genres-list">
                        {renderGenres(genres)}
                    </ul>
                </div>
                <time className="movie__release-date" dateTime={release_date}>
                    {release_date.slice(0, 4)}
                </time>
            </div>
        </Link>
  );
};

export default Movie;
