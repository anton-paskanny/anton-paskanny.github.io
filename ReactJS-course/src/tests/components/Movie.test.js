import React from 'react';
import { shallow } from 'enzyme';

import Movie from '../../client/components/Movie/Movie';

import { movieItem } from '../mocks/dataMock';

let component;
let props;

describe('Movie', () => {
    beforeEach(() => {
        props = movieItem;
        component = shallow(<Movie data={props} />);
    });
    it('render component correctly', () => {
        expect(component).toMatchSnapshot();
    });
    it('should contain Link with selected movie id', () => {
        expect(component.find('Link').prop('to')).toBe(`/movie/${props.id}`);
    });
});



