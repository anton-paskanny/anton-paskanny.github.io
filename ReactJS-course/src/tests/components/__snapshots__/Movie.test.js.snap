// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie render component correctly 1`] = `
<Link
  className="movie"
  replace={false}
  to="/movie/424785"
>
  <figure
    className="movie__img-placeholder"
  >
    <img
      alt="Transformers 7"
      className="movie__img"
      src="https://image.tmdb.org/t/p/w500/432BowXw7a4fWXSONxBaFKqvW4f.jpg"
    />
  </figure>
  <div
    className="movie__info-wrapper"
  >
    <div
      className="movie__desc-wrapper"
    >
      <h3
        className="movie__title"
      >
        Transformers 7
      </h3>
      <ul
        className="movie__genres-list"
      >
        <li
          className="movie__genre-list-item"
          key="Action"
        >
          Action
        </li>
        <li
          className="movie__genre-list-item--separator"
        >
          /
        </li>
        <li
          className="movie__genre-list-item"
        >
          Adventure
        </li>
      </ul>
    </div>
    <time
      className="movie__release-date"
      dateTime="2019-06-26"
    >
      2019
    </time>
  </div>
</Link>
`;
